src/index.ts

```diff
@@ -27,69 +35,168 @@ import {
 export
 namespace CommandIDs {
   export
-  const controlPanel: string = 'hub:control-panel';
+    const controlPanel: string = 'hub:control-panel';
+
+  export
+    const logout: string = 'hub:logout';
 
   export
-  const logout: string = 'hub:logout';
+    const restartServer: string = 'hub:restart-server';
 };
 
+const checkInterval: number = 7000;
 
-/**
- * Activate the jupyterhub extension.
- */
-function activateHubExtension(app: JupyterLab, palette: ICommandPalette, mainMenu: IMainMenu): void {
+export class HubExtension {
 
-  // This config is provided by JupyterHub to the single-user server app
-  // in a dictionary: app.web_app.settings['page_config_data'].
-  let hubHost = PageConfig.getOption('hub_host');
-  let hubPrefix = PageConfig.getOption('hub_prefix');
-  let baseUrl = PageConfig.getOption('baseUrl');
+  protected serverConnectionSettings: ServerConnection.ISettings;
+  protected app: JupyterLab;
+  protected status: string;
+  protected hubHost: string;
+  protected hubPrefix: string;
+  protected baseUrl: string;
+  protected restartUrl: string;
 
-  if (!hubPrefix) {
-    console.log('jupyterlab-hub: No configuration found.');
-    return;
-  }
+  constructor() {
+    this.serverConnectionSettings = ServerConnection.makeSettings();
+    this.status = 'ok';
 
-  console.log('jupyterlab-hub: Found configuration ',
-              {hubHost: hubHost, hubPrefix: hubPrefix});
+    // This config is provided by JupyterHub to the single-user server app
+    // in a dictionary: app.web_app.settings['page_config_data'].
+    this.hubHost = PageConfig.getOption('hub_host');
+    this.hubPrefix = PageConfig.getOption('hub_prefix');
```

Comment:
Should this be camelCase or snake_case? I vaguely remember seeing them being all camelCase, but could be wrong.